clio-client/src/main/scala/org/broadinstitute/clio/client/commands/ClioParsers.scala

```diff
@@ -22,13 +22,19 @@ trait ClioParsers {
   implicit def enumEntryParser[T <: EnumEntry: Enum](
     implicit c: ClassTag[T]
   ): ArgParser[T] = {
-    ArgParser.instance[T]("entry") { entry =>
-      val value = implicitly[Enum[T]].withNameOption(entry)
-      Either.cond(
-        value.isDefined,
-        value.get,
-        s"Unknown enum value $entry for type ${c.runtimeClass.getName} "
-      )
+    val enum = implicitly[Enum[T]]
+    val enumName = c.runtimeClass.getName
+    val entryMap = enum.namesToValuesMap
+    val entryString = entryMap.keys.mkString(",")
+
+    ArgParser.instance[T](entryString) { maybeEntry =>
+      entryMap
+        .get(maybeEntry)
+        .fold[Either[String, T]](
+          Left(
+            s"Unknown enum value '$maybeEntry' for type $enumName, valid values are [$entryString]"
+          )
+        )(Right.apply)
```

Comment:
Why couldn't this just be a Option.toRight with a left projection instead of a fold?

Reply:
Ooo, didn't know about that one! Will change.